pkg/volume/csi/csi_block.go

```diff
@@ -248,22 +248,58 @@ func (m *csiBlockMapper) SetUpDevice() (string, error) {
 	defer cancel()
 
 	// Call NodeStageVolume
-	stagingPath, err := m.stageVolumeForBlock(ctx, m.csiClient, accessMode, csiSource, attachment)
+	_, err = m.stageVolumeForBlock(ctx, m.csiClient, accessMode, csiSource, attachment)
 	if err != nil {
 		return "", err
 	}
 
+	return "", nil
+}
+
+func (m *csiBlockMapper) MapDevice(devicePath, globalMapPath, volumeMapPath, volumeMapName string, podUID types.UID) error {
+	// Get csiSource from spec
+	if m.spec == nil {
+		klog.Error(log("blockMapper.MapDevice spec is nil"))
+		return fmt.Errorf("spec is nil")
+	}
+
+	csiSource, err := getCSISourceFromSpec(m.spec)
+	if err != nil {
+		klog.Error(log("blockMapper.MapDevice failed to get CSI persistent source: %v", err))
+		return err
+	}
+
+	// Search for attachment by VolumeAttachment.Spec.Source.PersistentVolumeName
+	nodeName := string(m.plugin.host.GetNodeName())
+	attachID := getAttachmentName(csiSource.VolumeHandle, csiSource.Driver, nodeName)
+	attachment, err := m.k8s.StorageV1beta1().VolumeAttachments().Get(attachID, meta.GetOptions{})
+	if err != nil {
+		klog.Error(log("blockMapper.MapDevice failed to get volume attachment [id=%v]: %v", attachID, err))
+		return err
+	}
+
+	if attachment == nil {
+		klog.Error(log("blockMapper.MapDevice unable to find VolumeAttachment [id=%s]", attachID))
+		return errors.New("no existing VolumeAttachment found")
+	}
+
+	//TODO (vladimirvivien) implement better AccessModes mapping between k8s and CSI
+	accessMode := v1.ReadWriteOnce
+	if m.spec.PersistentVolume.Spec.AccessModes != nil {
+		accessMode = m.spec.PersistentVolume.Spec.AccessModes[0]
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), csiTimeout)
+	defer cancel()
+
 	// Call NodePublishVolume
+	stagingPath := m.getStagingPath()
```

Comment:
should this be getPublishPath? currently getPublishPath is unused

Reply:
Staging path is used [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_block.go#L190) for publish call, it is due to [CSI spec](https://github.com/container-storage-interface/spec/blame/master/spec.md#L2030).

It's confusing, so I agree that we should change ```publishVolumeForBlock``` not to take ```stagingPath``` as an argument, and get it inside the function, as it is done for [```publishPath```](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_block.go#L172).